src/index.ts

```diff
@@ -133,11 +152,20 @@ export default class Blockchain {
       this._common = new Common(chain, hardfork)
     }
 
+    if (opts.validate !== undefined) {
+      if (opts.validatePow !== undefined || opts.validateBlocks !== undefined) {
+        throw new Error(
+          "opts.validate can't be used at the same time than opts.validatePow nor opts.validateBlocks",
+        )
+      }
+    }
+
     // defaults
     this.db = opts.db ? opts.db : level()
     this.dbManager = new DBManager(this.db, this._common)
-    this.validate = opts.validate === undefined ? true : opts.validate
-    this.ethash = this.validate ? new Ethash(this.db) : null
+    this._validatePow = opts.validatePow !== undefined ? opts.validatePow : !!opts.validate
+    this._validateBlocks = opts.validateBlocks !== undefined ? opts.validateBlocks : !!opts.validate
```

Comment:
Sorry, thought a about this a bit longer and maybe it's exaggerated: but can we make this with an explicit `if (opts.validate) { ... } else { ... }` clause? I am especially afraid that this `true` default value for the two specific parameters gets too easily lost (if e.g. done by someone else) on the transition, since this is too implicitly side-integrated through `opts.validate`.

Reply:
Good idea, looks much better that way